server/publications/collections/products.js

```diff
@@ -308,47 +341,24 @@ Meteor.publish("Products", function (productScrollLimit = 24, productFilters, so
 
   // TODO: Consider publishing the non-admin publication if a user is not in "edit mode" to see what is published
 
-  // Active shop
-  const shopId = Reaction.getShopId();
-  const primaryShopId = Reaction.getPrimaryShopId();
-
-  // Get a list of shopIds that this user has "createProduct" permissions for (owner permission is checked by default)
-  const userAdminShopIds = Reaction.getShopsWithRoles(["createProduct"], this.userId);
-
-  // Don't publish if we're missing an active or primary shopId
-  if (!shopId || !primaryShopId) {
-    return this.ready();
-  }
-
-  // Get active shop id's to use for filtering
-  const activeShopsIds = Shops.find({
-    $or: [
-      { "workflow.status": "active" },
-      { _id: primaryShopId }
-    ]
-  }, {
-    fields: {
-      _id: 1
-    }
-  }).fetch().map((activeShop) => activeShop._id);
-
   const selector = filterProducts(productFilters);
 
   if (selector === false) {
     return this.ready();
   }
 
+  // Get a list of shopIds that this user has "createProduct" permissions for (owner permission is checked by default)
+  const userAdminShopIds = Reaction.getShopsWithRoles(["createProduct"], this.userId);
+
   // We publish an admin version of this publication to admins of products who are in "Edit Mode"
   // Authorized content curators for shops get special publication of the product
   // with all relevant revisions all is one package
   // userAdminShopIds is a list of shopIds that the user has createProduct or owner access for
-  if (editMode && userAdminShopIds && Array.isArray(userAdminShopIds) && userAdminShopIds.length > 0) {
+  if (editMode && Array.isArray(userAdminShopIds) && userAdminShopIds.length > 0) {
     selector.isVisible = {
       $in: [true, false, null, undefined]
     };
-    selector.shopId = {
-      $in: userAdminShopIds
-    };
+    selector.shopId.$in = _.intersection(selector.shopId.$in, userAdminShopIds);
```

Comment:
the intersection of "accessible shops", "requested shops", and "shops for which I am an admin"

Reply:
Are you sure we want to see products that can't be edited when `editMode` is on? Might that be confusing? Or we just publish them all and let the client UI indicate which are editable? @spencern or @rymorgan

Reply:
the current behavior on master is to show them all, which I agree, is confusing.
this line will omit shops for which you do not have edit rights.
I think we are saying the same thing.

further, I think your point about deleting `isVisible` is correct here too